backend/routes/data.py

```diff
@@ -71,6 +120,31 @@ def add_data():
         db.session.add(data)
         db.session.commit()
         db.session.refresh(data)
+
+        if isinstance(segmentations, str):
+            segmentations = json.loads(segmentations)
+        elif segmentations != []:
+            app.logger.error("Could not parse segmentations from ",
+                            type(segmentations))
+        annotations = []
+        for segment in segmentations:
+            validation = validate_segmentation(segment)
+            if validation:
+                segmentation = generate_segmentation(
+                    data_id=data.id,
+                    end_time=segment['end_time'],
+                    start_time=segment['start_time'],
+                    annotations=segment['annotations'],
+                    transcription=segment['transcription'],
+                )
+
+                db.session.add(segmentation)
```

Comment:
We should commit only after all segmentations are created and validated.

Reply:
I will recommend creating a list of segmentations and assign them to datapoint as `data.segmentations = segmentations` and commit datapoint only.

Reply:
I thought  since segmentations are supposed to have a `backref` to `Data` via a data_id. We cannot create a  `Segmentation` without assigning the `data_id`. Coming commit will have the requested changes.

Reply:
@DumbMachine Check if it works. I believe it works without `data_id`.